### PR TITLE
FIT-237 Add Hooks for Smoke Tests in Jenkins File for Dashboard Proje…

### PR DESCRIPTION
**Pull request by mariamcws**
…ct, refactor

## Description
Add Hooks for Smoke Tests in Jenkins File for Dashboard Project and make it available for other projects via jenkins pipeline
## Jira Issue link

[Add Hooks for Smoke Tests in Jenkins File for Dashboard Project](https://osi-cwds.atlassian.net/browse/FIT-237)

## Tests
- [x] I have included unit tests

### CODE REVIEW

File: test/gov/ca/cwds/jenkins/SmokeTesterSpecification.groovy [hunk 1]
```diff
@@ -8,32 +8,37 @@ class SmokeTesterSpecification extends Specification {
...
     def PipeLineScript() {}
   }
 
   def "#runSmokeTestPasses"() {
 
     given:
-    def pipeline = Mock(PipeLineScript)
-    def passed = './test/resources/smoketest/passed.sh'
+    def PipeLineScript pipeline = Mock(PipeLineScript)
+    pipeline.sh (_) >> "smoketest passed"
```
Comment (edgibbs) [inline test/gov/ca/cwds/jenkins/SmokeTesterSpecification.groovy]: So just to be explicit about matching the arguments can we add:

```groovy
 pipeline.sh([script: "passed test", returnStdout: true]) >> "smoketest passed"
```

File: test/gov/ca/cwds/jenkins/SmokeTesterSpecification.groovy [hunk 2]
```diff
@@ -8,32 +8,37 @@ class SmokeTesterSpecification extends Specification {
...
+    testResult == "smoke test passed"
   }
 
   def "#runSmokeTestFails"() {
 
     given:
     def pipeline = Mock(PipeLineScript)
-    def failed = './test/resources/smoketest/failed.sh'
     def smokeTester = new SmokeTester(pipeline)
+    pipeline.sh (_) >> "smoketest failed"
```
Comment (edgibbs) [inline test/gov/ca/cwds/jenkins/SmokeTesterSpecification.groovy]: same here with the arguments.

File: test/gov/ca/cwds/jenkins/SmokeTesterSpecification.groovy (hunk 1 above)
Comment (mariamcws) [reply test/gov/ca/cwds/jenkins/SmokeTesterSpecification.groovy]: fixed

File: test/gov/ca/cwds/jenkins/SmokeTesterSpecification.groovy (hunk 2 above)
Comment (mariamcws) [reply test/gov/ca/cwds/jenkins/SmokeTesterSpecification.groovy]: fixed

**Pull request merged**